### PR TITLE
Make it possible to search for the source code as if the patch was applied

### PR DESCRIPTION
**Pull request by boiko**
This is useful when verifying if a new release of the software has the fix applied upstream.

### CODE REVIEW

File: src/main/java/com/suse/pase/query/QueryFactory.java [hunk 1]
```diff
@@ -40,6 +38,23 @@
...
+
+        return ud.getFiles().stream()
+                .filter(f -> !f.getFromFile().equals("/dev/null"))
+                .map(file ->{
+                    var path = file.getFromFile();
+                    var chunks = file.getPatch().getDeltas().stream()
+                            .filter(d -> of(INSERT, EQUAL, CHANGE).contains(d.getType()))
+                            .map(d -> d.getTarget().getLines().stream().filter(l -> !l.isEmpty()).collect(toList()))
+                            .collect(toList());
+                    return new PatchTargetQuery(path, chunks);
```
Comment (moio) [inline src/main/java/com/suse/pase/query/QueryFactory.java]:
```suggestion
                    return new PatchQuery(path, chunks);
```

Alternate class name suggestion, to make clear there are two possible cases for this query (patch target or applied patch).

PS. I recommend to use an IDE to do this kind of refactoring (renaming) in a safe way.

I highly recomment IntelliJ IDEA if you are not using it already.

Comment (moio) [reply src/main/java/com/suse/pase/query/QueryFactory.java]: Once the class is renamed to `PatchQuery`, I'd DRY up `buildAppliedPatchTargetQuery` and `buildPatchTargetQuery` by creating a generic `buildPatchQuery` which takes a `List<DeltaType>` as parameter.

Then I can imagine implementing `buildAppliedPatchTargetQuery` by calling `buildPatchQuery` with `of(DELETE, EQUAL, CHANGE)` and `buildAppliedPatchTargetQuery` with `of(INSERT, EQUAL, CHANGE)`.

Finally I would recommend renaming `buildAppliedPatchTargetQuery` to `buildAppliedPatchQuery` as "target" does not make too much (English) sense at that point.

File: src/test/java/com/suse/pase/MainTest.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ static void setup() throws Exception {
     @org.junit.jupiter.api.Test
     public void searchTest() throws Exception {
         var patchPath = resourcePath.resolve("patches").resolve("CVE-2017-5638.patch");
-        var results = Search.search(indexPath, patchPath, false, false);
+        var results = Search.search(indexPath, patchPath, false, false, false);
```
Comment (moio) [inline src/test/java/com/suse/pase/MainTest.java:38]: Ideally we should add tests for the new query types - better but not necessarily in this PR, just eventually.

**Review by moio — COMMENTED** [review]
It looks pretty good already! I only have a couple of suggestions which are mostly cosmetic/stylish in their nature.

Then this is ready to merge (as soon as you tested that it works! :wink:)

Good job!

File: src/test/java/com/suse/pase/MainTest.java (hunk 2 above)
Comment (boiko) [reply src/test/java/com/suse/pase/MainTest.java:38]: I will try to write some in a separate pull request, let's see how that goes.

**Review by moio — APPROVED** [review]
LGTM. If even more types are ever to be added, one option is to use an enumeration instead of passing even more booleans around. But that's style nitpicking by and large!

**Pull request merged**